File: src/commands/status.ts
import { loadConfig } from '../core/config.js';
import { getDb, todayCount, recentLikes, lastLogin } from '../core/db.js';
import chalk from 'chalk';

export async function status(): Promise<void> {
  const config = loadConfig();
  const db = getDb();

  const today = todayCount(db);
  const recent = recentLikes(db, 10);
  const lastLoginAt = lastLogin(db);

  console.log(chalk.bold('forge-linkedin v0.2.0'));
  console.log('─'.repeat(32));
  console.log(`Today's likes : ${chalk.bold(today)} / ${config.daily_cap}`);
  console.log(`Last login    : ${lastLoginAt ? chalk.green(lastLoginAt) : chalk.gray('never')}`);
  console.log(`Last run      : ${recent.length > 0 ? chalk.green(recent[0].liked_at) : chalk.gray('never')}`);

  if (recent.length > 0) {
    console.log('\nRecent likes:');
    recent.forEach((p, i) => {
      console.log(`  ${i + 1}. ${chalk.bold(p.author)} — ${p.hashtag} — ${chalk.gray(p.liked_at)}`);
    });
  } else {
    console.log('\nRecent likes: (none)');
  }
}

File: crates/forge-linkedin/src/commands/status.rs
use anyhow::{bail, Result};
use std::path::Path;

pub async fn run(forge_dir: &Path) -> Result<()> {
    let db_path = forge_dir.join("data.db");
    if !db_path.exists() {
        bail!("No database found. Run `forge-linkedin init` first.");
    }
    let store = store::Store::open(&db_path)?;

    let config_path = forge_dir.join("config.json");
    let cap = if config_path.exists() {
        let raw = std::fs::read_to_string(&config_path)?;
        let cfg: linkedin_ops::AppConfig = serde_json::from_str(&raw).unwrap_or_default();
        cfg.daily_cap
    } else {
        30
    };

    let today_count = store.today_count()?;
    let skipped = store.today_skip_count()?;
    let last_login = store.last_login()?;
    let recent = store.recent_likes(5)?;

    println!("=== forge-linkedin status ===");
    println!("Today: {} / {} likes", today_count, cap);
    println!("Today skipped: {}", skipped);
    println!(
        "Last login: {}",
        last_login.as_deref().unwrap_or("never")
    );
    println!();
    if recent.is_empty() {
        println!("No likes recorded yet.");
    } else {
        println!("Last {} likes:", recent.len());
        for post in &recent {
            println!(
                "  {} | {} | {} | {}",
                post.liked_at.format("%Y-%m-%d %H:%M"),
                post.hashtag,
                post.author,
                post.post_id
            );
        }
    }

    Ok(())
}
